### PR TITLE
OSD-2417 letsencrypt maintenance error metric

### PR DESCRIPTION
**Pull request by yithian**

### CODE REVIEW

File: hack/test/local_test.sh [hunk 1]
```diff
@@ -85,8 +85,11 @@ mkdir ${tmpdir}
 
 # Start minikube with extra memory (needed for the go build)
 MINIKUBE_OPTIONS="--memory=6g --bootstrapper=kubeadm --extra-config=kubelet.authentication-token-webhook=true --extra-config=kubelet.authorization-mode=Webhook --extra-config=scheduler.address=0.0.0.0 --extra-config=controller-manager.address=0.0.0.0 --extra-config=apiserver.service-node-port-range=1-65535"
+# kubernetes v1.22 stopped supporting CustomResourceDefinition in apiextensions.k8s.io/v1beta1
+# unfortunately, the router CRDs are still using that API
```
Comment (cblecker) [inline hack/test/local_test.sh:89]: which CRDs? I thought we had already dealt with those kube api deprecations

File: pkg/controller/certificaterequest/issue_certificate.go [hunk 2]
```diff
@@ -34,10 +35,14 @@ import (
 	"github.com/openshift/certman-operator/pkg/localmetrics"
 )
 
+const (
+	leMaintMessage = "The service is down for maintenance or had an internal error."
```
Comment (cblecker) [inline pkg/controller/certificaterequest/issue_certificate.go:39]: Is there something like an HTTP status code that could give us this status without relying on string matching? (this is obv way better than nothing, but wondering if there is something programmatic that is more robust)

File: hack/test/local_test.sh (hunk 1 above)
Comment (yithian) [reply hack/test/local_test.sh:89]: from [router](https://github.com/openshift/router/):
```
$ kubectl create -n openshift-ingress -f deploy/route_crd.yaml
error: unable to recognize "deploy/route_crd.yaml": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
```
```
$ kubectl create -n openshift-ingress -f deploy/router_rbac.yaml
clusterrolebinding.rbac.authorization.k8s.io/openshift-ingress-router created
clusterrolebinding.rbac.authorization.k8s.io/openshift-ingress-router-auth-delegator created
namespace/openshift-ingress created
serviceaccount/ingress-router created
error: unable to recognize "deploy/router_rbac.yaml": no matches for kind "ClusterRole" in version "rbac.authorization.k8s.io/v1beta1"
```
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"archive", BuildDate:"2021-10-28T06:55:39Z", GoVersion:"go1.17.2", Compiler:"gc", Platform:"linux/
amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:32:41Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/am
d64"}
```

File: pkg/controller/certificaterequest/issue_certificate.go (hunk 2 above)
Comment (yithian) [reply pkg/controller/certificaterequest/issue_certificate.go:39]: I'm honestly not sure. The output captured in [OSD-8663](https://issues.redhat.com/browse/OSD-8663) suggests maybe it's error code 0, but that doesn't seem right to me? I'm not sure how to go about finding out what error code might be generated by the server, either. Maybe search through any open source ACME server code?

**Comment by yithian** [comment]
@cblecker @dustman9000 any changes to this you'd like to see?

**Comment by dustman9000** [comment]
Looks good to me, Christoph did have a question about checking an error code that I think we can address later when we know more about LE response codes.
/lgtm

**Pull request merged**